Sprint 9 solution http api

isakovleonid opened the pull request.

isakovleonid committed:
Финальное задания Спринта 9

1) Реализован класс для http-сервера
2) Реализованы частично методы для endpoint tasks

isakovleonid committed:
Финальное задания Спринта 9

1) Реализован класс для http-сервера
2) Реализованы обработчики endpoints
3) Доработаны тест-кейсы

isakovleonid committed:
Финальное задания Спринта 9

1) Реализован класс для http-сервера
2) Реализованы обработчики endpoints
3) Доработаны тест-кейсы

Alexander-Bolotov commented on src/ru/isakovleonid/practicum/taskmanager/httpserver/SubTasksHandler.java:
```diff
@@ -0,0 +1,111 @@
+
+                        handleDeleteSubTaskId(id);
+
+                        super.sendCreated(exchange, "Задача удалена");
```
при удалении должен быть код ответ 200 (по ТЗ)

isakovleonid replied:
исправлено

Alexander-Bolotov commented on src/ru/isakovleonid/practicum/taskmanager/httpserver/TasksHandler.java:
```diff
@@ -0,0 +1,112 @@
+
+                        handleDeleteTaskId(id);
+
+                        super.sendCreated(exchange, "Задача удалена");
```
должно быть 200

isakovleonid replied:
исправлено

Alexander-Bolotov commented on src/ru/isakovleonid/practicum/taskmanager/httpserver/BaseHttpHandler.java:
```diff
@@ -0,0 +1,68 @@
+        exchange.close();
+    }
+
+    protected void sendHasInteractions(HttpExchange exchange, String text) throws IOException {
```
нигде не используется, значит пересечение не проверяется?

isakovleonid replied:
У меня нет возможности пересечений - при передачи id в запросе произойдет обновление

Alexander-Bolotov commented on test/ru/isakovleonid/practicum/taskmanager/httpserver/HttpTaskServerTest.java:
```diff
@@ -0,0 +1,399 @@
+        assertEquals(404, response.statusCode(), "Некорректный статус запроса при поытке удаления несуществующей подзадачи");
+    }
+}
+
```
нужно добавить проверку обработки пересечений задач и возврата 406​

isakovleonid replied:
пересечение - это когда пришел запрос на обновление с id, который сейчас принадлежит задаче другого типа?

isakovleonid committed:
Финальное задания Спринта 9

Замечания по codereview

isakovleonid committed:
Финальное задания Спринта 9

Замечания по codereview

isakovleonid committed:
Финальное задания Спринта 9

Замечания по codereview.
1) Добавлена обработка пересечений.
2) Добавлены тест-кейсы для пересечений
